capsules/src/rng.rs

```diff
@@ -29,24 +36,28 @@ pub struct App {
     idx: usize,
 }
 
-pub struct SimpleRng<'a, RNG: rng::RNG> {
-    rng: &'a RNG,
+pub struct RngDriver<'a> {
+    rng: &'a Rng<'a>,
     apps: Grant<App>,
     getting_randomness: Cell<bool>,
 }
 
-impl<RNG: rng::RNG> SimpleRng<'a, RNG> {
-    pub fn new(rng: &'a RNG, grant: Grant<App>) -> SimpleRng<'a, RNG> {
-        SimpleRng {
+impl<'a> RngDriver<'a> {
```

Comment:
Applies to all `impls`, you can remove in-band lifetimes (lifetime annotations after impl):
imp RngDriver<'a>

Reply:
Removed.